enhancements/cluster-logging/hypershift-audit-logs.md

```diff
@@ -0,0 +1,219 @@
+
+### Implementation Details
+
+#### ClusterLogForwarder multiple instances
+
+The ClusterLogForwarder is currently a singleton.
+To provide isolation between HCPs we need to deploy a forwarder per HCP namespace.
+This requires modifications to the CLO 
+
+_FIXME JIRA ref._
+
+#### ClusterLogForwarder HTTP listener
+
+The ClusterLogForwarder needs to be extended to act a a HTTP listener,
+so it can receive logs as a web hook from the API server in the HCP.
+
+_FIXME JIRA reference_
+
+_FIXME Is there a way to use the file system and avoid this requirement?_
+
+#### Hypershift Logging Operator
+
+This operator must
+- watch resources in a HCP API server, update their status 
+  - `HypershiftLogForwarder` 
+  - `Policy.audit.k8s.io`
+- create resources in the management cluster API server
+  - `ClusterLogForwarder`
+  - `Policy.audit.k8s.io`
+
+The policy objects can be copied verbatim. 
+
+The `HypershiftLogForwarder` fields are compatible with `ClusterLogForwarder` fields but some changes
+and validation may be needed:
+- validate that the forwarding request is valid and allowed.
+- possibly replace named outputs with a suitable output configuration;
+  if we want to restrict the user to a fixed set of alternatives
+- possibly add credentials derived from the guest cluster or the management cluster.
+  
+### Risks and Mitigations
+
+CPU/Network cost of extra log forwarding
+- We have to do this anyway, but user might choose a more verbose policy.
+- We could restrict the available logs at the API server to control this.
+
+### Drawbacks
+
+## Design Details
+
+### Open Questions
+
+- Is there any way to write the audit logs to a file system that the log forwarder can read?
+  This would be simpler & more reliable than a web hook.
+- Can an operator on the management cluster reconcile and update status on resources in the guest cluster directly?
+  The API server is running in the HCP on the management cluster, but need to check for possible snags.
+- Should we allow the user to specify any destination for logs from the management cluster? If not:
+   - Why not? We allow that freedom on self contained clusters. I see no new security/privacy issues.
+   - We can instead provide a fixed set of named alternatives. The `HyperShiftLoggingOperator`
+     can translate those names into known configurations.
+- How do we manage credentials? Assuming they are stored in the HCP API server, can the management cluster
+  access secrets for TLS security and the like? If not how do we get at them.
+
+### Test Plan
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+#### Tech Preview -> GA
+
+#### Removing a deprecated feature
+
+None.
+
+### Upgrade / Downgrade Strategy
+
+Nothing new.
+
+### Version Skew Strategy
+
+Nothing new.
+
+### Operational Aspects of API Extensions
+
+An operator on the management server that watches objects on the guest cluster is new (is it?)
```

Comment:
It's not new depending on who you ask :) 
SRE have several operators running in each HCP and there are the HCP components themselves, which run as pods on the MC and connect to the HCP API.

Reply:
Glad to hear it! I felt like it should be possible but didn't know if it had been done.